qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -81,75 +92,75 @@ def __init__(
         self.include_custom = include_custom
         self.ansatz = ansatz
         self.initial_state = initial_state
-        self._vqe = VQE(
-            ansatz=None,
-            quantum_instance=self._quantum_instance,
-            optimizer=self._optimizer,
-            initial_point=self._initial_point,
-            gradient=self._gradient,
-            expectation=self._expectation,
-            include_custom=self._include_custom,
-        )
+        self.callback = callback
 
     @property
     def quantum_instance(self) -> QuantumInstance:
         """Getter of the quantum instance."""
-        return self._quantum_instance
+        return self._vqe.quantum_instance
 
     @quantum_instance.setter
     def quantum_instance(self, q_instance: QuantumInstance) -> None:
         """Setter of the quantum instance."""
-        self._quantum_instance = q_instance
+        self._vqe.quantum_instance = q_instance
 
     @property
     def optimizer(self) -> Optional[Optimizer]:
         """Getter of the optimizer."""
-        return self._optimizer
+        return self._vqe.optimizer
 
     @optimizer.setter
     def optimizer(self, optimizer: Optional[Optimizer]) -> None:
         """Setter of the optimizer."""
-        self._optimizer = optimizer
+        if optimizer is None:
+            return
+        self._vqe.optimizer = optimizer
 
     @property
     def initial_point(self) -> Optional[np.ndarray]:
         """Getter of the initial point."""
-        return self._initial_point
+        return self._vqe.initial_point
 
     @initial_point.setter
     def initial_point(self, initial_point: Optional[np.ndarray]) -> None:
         """Setter of the initial point."""
-        self._initial_point = initial_point
+        if initial_point is None:
+            return
+        self._vqe.initial_point = initial_point
 
     @property
     def gradient(self) -> Optional[Union[GradientBase, Callable]]:
         """Getter of the gradient function"""
-        return self._gradient
+        return self._vqe.gradient
 
     @gradient.setter
     def gradient(self, gradient: Optional[Union[GradientBase, Callable]]) -> None:
         """Setter of the gradient function"""
-        self._gradient = gradient
+        if gradient is None:
+            return
```

Comment:
Actually the setters of VQE for gradient and optimizer do not revert to default behavior when passing in `None`. Instead, the set the internal object to `None` which leads to errors further down the line. I agree that this is not optimal but this needs to be fixed on Terra's side (see next comment below)

Reply:
Either way passing in None here is in essence ignored which maybe is not expected either and arguably just as wrong. Why not pass it on, that way when VQE is corrected this will work as expected.